### PR TITLE
Adding ClamAv Lib

### PR DESCRIPTION
**Pull request by alexjoybc**
A new libs for supporting clamAV

### CODE REVIEW

File: src/spring-clamav-starter/README.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        <groupId>ca.bc.gov.open</groupId>
+        <artifactId>spring-clamav-starter</artifactId>
+        <version>0.1.4</version>
+    </dependency>
+</dependencies>
+```
+
+## How to use the service
+
+Code is self explanatory. Something like this is the idea:
```
Comment (akroon3r) [inline src/spring-clamav-starter/README.md:21]:
```suggestion
The following code should be used to implement this component:
```

File: src/spring-clamav-starter/pom.xml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+                            <goal>report</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+
+
+</project>
```
Comment (akroon3r) [inline src/spring-clamav-starter/pom.xml:102]: newline :O

File: src/spring-clamav-starter/src/main/java/ca/bc/gov/open/clamav/starter/AutoConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    public AutoConfiguration(ClamAvProperties clamAvProperties) {
+        this.clamAvProperties = clamAvProperties;
+    }
+
+    @Bean
+    @ConditionalOnMissingBean(ClamAVClient.class)
+    public ClamAVClient clamAVClient() {
+
+        logger.debug("Configuring ClamAv Client");
```
Comment (akroon3r) [inline src/spring-clamav-starter/src/main/java/ca/bc/gov/open/clamav/starter/AutoConfiguration.java:25]: debug or info?

**Comment by akroon3r** [comment]
Other than newlines on generated files this is fine.

File: src/spring-clamav-starter/src/main/java/ca/bc/gov/open/clamav/starter/AutoConfiguration.java (hunk 3 above)
Comment (alexjoybc) [reply src/spring-clamav-starter/src/main/java/ca/bc/gov/open/clamav/starter/AutoConfiguration.java:25]: yeah debug is right here

File: src/spring-clamav-starter/pom.xml (hunk 2 above)
Comment (alexjoybc) [inline src/spring-clamav-starter/pom.xml:102]:
```suggestion
</project>

```

File: src/spring-clamav-starter/src/main/resources/META-INF/spring.factories [hunk 4]
```diff
@@ -0,0 +1,3 @@
+org.springframework.boot.autoconfigure.EnableAutoConfiguration=\
+  ca.bc.gov.open.clamav.starter.AutoConfiguration
+
```
Comment (alexjoybc) [inline src/spring-clamav-starter/src/main/resources/META-INF/spring.factories:3]:
```suggestion

```

File: src/spring-clamav-starter/src/test/java/ca/bc/gov/open/clamav/starter/AutoConfigurationTest.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    @Test
+    void clamAVClient() {
+        context.run(it -> { assertThat(it).hasSingleBean(ClamAVClient.class); });
+    }
+
+    @Test
+    void clamAvService() {
+        context.run(it -> { assertThat(it).hasSingleBean(ClamAvService.class); });
+    }
+}
```
Comment (alexjoybc) [inline src/spring-clamav-starter/src/test/java/ca/bc/gov/open/clamav/starter/AutoConfigurationTest.java:41]:
```suggestion
}

```

File: src/spring-clamav-starter/src/test/java/ca/bc/gov/open/clamav/starter/ClamAvServiceImplTest.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    @Test
+    @DisplayName("OK: test ping")
+    public void testPing() throws IOException {
+
+        Mockito.when(clamAvClient.ping()).thenReturn(true);
+
+        Assertions.assertTrue(sut.ping());
+    }
+
+}
```
Comment (alexjoybc) [inline src/spring-clamav-starter/src/test/java/ca/bc/gov/open/clamav/starter/ClamAvServiceImplTest.java:67]:
```suggestion
}

```

**Pull request merged**